### PR TITLE
[Harness] Add AppRunner.RunAsync unit tests

### PR DESCRIPTION
**Pull request by premun**
Adds the most important batch of unit tests for executing test runs

### CODE REVIEW

File: tests/xharness/AppRunner.cs [hunk 1]
```diff
@@ -655,26 +659,27 @@ public async Task<int> RunAsync ()
...
+					await crashReporter.StartCaptureAsync ();
 
 					MainLog.WriteLine ("Starting test run");
 
 					bool waitedForExit = true;
 					// We need to check for MT1111 (which means that mlaunch won't wait for the app to exit).
 					var callbackLog = new CallbackLog ((line) => {
 						// MT1111: Application launched successfully, but it's not possible to wait for the app to exit as requested because it's not possible to detect app termination when launching using gdbserver
-						waitedForExit &= line?.Contains ("MT1111: ") != true;
+						waitedForExit &= !line?.Contains ("MT1111: ") != true;
```
Comment (rolfbjarne) [inline tests/xharness/AppRunner.cs]: This looks strange, what's the reason for this change?

Comment (premun) [reply tests/xharness/AppRunner.cs]: Oops, my bad, will revert

File: tests/xharness/Xharness.Tests/Tests/AppRunnerTests.cs [hunk 2]
```diff
@@ -300,10 +291,479 @@ public async Task UninstallFromDeviceTest ()
...
+				ips.Append (ipAddresses [i].ToString ());
+			}
+
+			var expectedArgs = $"--sdkroot {xcodePath} -v -v -argument=-connection-mode -argument=none " +
+				$"-argument=-app-arg:-autostart -setenv=NUNIT_AUTOSTART=true -argument=-app-arg:-autoexit " +
+				$"-setenv=NUNIT_AUTOEXIT=true -argument=-app-arg:-enablenetwork -setenv=NUNIT_ENABLE_NETWORK=true " +
+				$"-setenv=DISABLE_SYSTEM_PERMISSION_TESTS=1 -argument=-app-arg:-hostname:{ips} -setenv=NUNIT_HOSTNAME={ips} " +
+				$"-argument=-app-arg:-transport:Tcp -setenv=NUNIT_TRANSPORT=TCP -argument=-app-arg:-hostport:{simpleListener.Object.Port} " +
+				$"-setenv=NUNIT_HOSTPORT={simpleListener.Object.Port} -setenv=env1=value1 -setenv=env2=value2 " +
+				$"--launchdev {appPath} --disable-memory-limits --wait-for-exit --devname Test iPad";
```
[the remark this answers is not in the document]
Comment (mandel-macaque) [reply tests/xharness/Xharness.Tests/Tests/AppRunnerTests.cs]: yes, makes sense, and it was like that already. Looks good do me.
